doc/getting-started/building-from-source.rst

```diff
@@ -172,103 +178,68 @@ Assuming that you are at the top level of the acrn-hypervisor directory, perform
+  scenario XML files are ``/path/to/board.xml`` and ``/path/to/scenario.xml``:
 
-   # Modify the configurations per your needs
-   $ cd ../         # Enter top-level folder of acrn-hypervisor source
-   $ make menuconfig -C hypervisor
-   # modify your own "ACRN Scenario" and "Target board" that want to build
-   # in pop up menu
+  .. code-block:: none
 
-Note that ``menuconfig`` is python3 only.
+    $ make all BOARD=/path/to/board.xml SCENARIO=/path/to/scenario.xml
 
-Refer to the help on menuconfig for a detailed guide on the interface:
+.. note::
+   ACRN uses XML files to summarize board characteristics and scenario
+   settings. The ``BOARD`` and ``SCENARIO`` variables accept board/scenario
+   names as well as paths to XML files. When board/scenario names are given, the
+   build system searches for XML files with the same names under
+   ``misc/config_tools/data/``. When paths (absolute or relative) to the XML
+   files are given, the build system uses the files pointed at. If relative
+   paths are used, they are considered relative to the current working
+   directory.
+
+See the :ref:`hardware` document for information about platform needs for each
+scenario. For more instructions to customize scenarios, see
+:ref:`getting-started-hypervisor-configuration` and
+:ref:`acrn_configuration_tool`.
 
-.. code-block:: none
+The build results are found in the ``build`` directory. You can specify
+a different build directory by setting the ``O`` ``make`` parameter,
+for example: ``make O=build-nuc``.
 
-   $ pydoc3 menuconfig
+.. _getting-started-hypervisor-configuration:
 
 .. rst-class:: numbered-step
 
-Build the Hypervisor, Device Model, and Tools
-*********************************************
-
-Now you can build all these components at once as follows:
-
-.. code-block:: none
-
-   $ make 	# Build hypervisor with the new .config
+Modify the Hypervisor Configuration
+***********************************
 
-The build results are found in the ``build`` directory. You can specify
-a different Output folder by setting the ``O`` ``make`` parameter,
-for example: ``make O=build-nuc``.
+The ACRN hypervisor is built with scenario encoded in an XML file (referred to
+as the scenario XML hereinafter). The scenario XML of a build can be found at
+``<build>/hypervisor/.scenario.xml``, where ``<build>`` is the name of the build
+directory. You can make further changes to this file to adjust to your specific
+requirements. Another ``make`` will rebuild the hypervisor using the updated
+scenario XML.
 
+The following commands show how to customize the scenario XML based on the
+predefined ``INDUSTRY`` scenario for ``nuc7i7dnb`` and rebuild the hypervisor.
 
 .. code-block:: none
 
-   $ make all BOARD_FILE=$PWD/misc/vm_configs/xmls/board-xmls/nuc7i7dnb.xml \
-   SCENARIO_FILE=$PWD/misc/vm_configs/xmls/config-xmls/nuc7i7dnb/industry.xml TARGET_DIR=xxx
+   $ make all BOARD=nuc7i7dnb SCENARIO=industry RELEASE=0
```

Comment:
The default now is to build a debug version.  If you do want to leave the RELEASE setting on this make command, does it need to be ``RELEASE=n``?

Reply:
I'll remove the `RELEASE=0` setting for simplicity. The debug/release setting does not make any difference here as we want to introduce the steps to customize configurations.

Reply:
I believe `all` is also the target by default and I would therefore recommend that we leave it out and simply use `make BOARD=xxx SCENARIO=xxx` throughout the documents.

We could indicate a particular component (e.g. `hypervisor`, `devicemodel`, `doc`) if and where we'd want them to build just a particular component. **Although** I am actually wondering if the `RELEASE` value would be properly handled if one wants to build the `devicemodel` or `tools` only for example (as the logic for assigning the value to `RELEASE` lies in `hypervisor/scripts/makefile/`

Reply:
I'll remove the unnecessary `all` throughout the guide.

I think the inconsistency between the top-level Makefile and the component-wise Makefiles has already made it subtle. I'll try creating a PR to fix this first.

Reply:
Thanks @junjiemao1 👍